docs/dependencies.md

```diff
@@ -0,0 +1,286 @@
+
+One case where we apply added scrutiny to third-party dependencies is
+JS libraries.  They are a particularly important concern because we
+want to keep the Zulip webapp's JS bundle small, so that Zulip
+continues to load quickly on systems with low network bandwidth.
+We'll look at large JS libraries with much greater scrutiny for
+whether their functionality justifies their size than Python
+dependencies, since an extra 50KB of code usually doesn't matter in
+the backend, but does in JavaScript.
+
+## System packages
+
+For the third-party services like Postgres, Redis, Nginx, and RabbitMQ
+that are documented in the
+[architecture overview](architecture-overview.html), we rely on the
+versions of those packages provided alongside the Linux distribution
+on which Zulip is deployed (for Debian and Ubuntu, this means `apt`).
+Since we don't control the versions of these dependencies, we avoid
+relying on specific versions of these packages wherever possible.
+
+The exact lists of `apt` packages needed by Zulip are maintained in
+two major places:
+* For production, in our puppet configuration, `puppet/zulip/`, using
+  the `Package` and `SafePackage` directives.  For development, in
+  `APT_DEPENDENCIES` in `tools/lib/provision.py`.
+* The packages needed to build a Zulip virtualenv, in
+  `VENV_DEPENDENCIES`.  These are separate from the rest because we
+  may need to install a virtualenv before running the more complex
+  scripts that, in turn, install other dependencies, and also because
+  that list is shared between development and production.
+
+We maintain a [personal package archive (PPA)][ppa] with some packages
+unique to Zulip (e.g the `tsearch_extras` postgres extension) and
+backported versions of other dependencies (e.g. `camo`, to fix a buggy
+`init` script).  Our goal is to shrink or eliminate this PPA where
+possible by getting issues addressed in the upstream distributions.
+
+We also rely on the `pgroonga` PPA for the `pgroonga` postgres
+extension, used by our [full-text search][full-text-search.html].
+
+## Python packages
+
+We manage Python packages via the Python-standard `requirements.txt`
+system and virtualenvs, but there’s a number of interesting details
+about how Zulip makes this system work well for us that are worth
+highlighting.  The system is largely managed by the code in
+`scripts/lib/setup_venv.py`
+
+* **Using `pip` to manage dependencies**.  This is standard in the
+  Python ecosystem, and means we only need to record a list of
+  versions in a `requirements.txt` file to declare what we're using.
+  Since we have a few different installation targets, we maintain
+  several `requirements.txt` format files in the `requirements/`
+  directory (e.g. `dev.txt` for development, `prod.txt` for
+  production, `docs.txt` for ReadTheDocs, `common.txt` for the vast
+  major of packages common to prod and development, etc.).  We use
+  with `pip --no-deps` to ensure that we're declaring dependencies
```

Comment:
maybe something like 
"We use `--no-deps` with `pip` to ensure that we're declaring dependencies explicitly."
or just remove "with" from current sentence.